### ISSUE TITLE
--maxsleep broken for values less than 1000

### ISSUE THREAD

**Issue by GrantEdwards** [appneta/tcpreplay#421]
On Linux, the --maxsleep option only seems to work correctly for values of 1000 or larger.  [Correction, it's broken for any value that isn't a multiple of 1000.]  Values smaller than 1000 send packets _way_ faster than they should.  It's hard to get accurate test results, but it sort of looks like values less than 1000 are getting treated as microseconds rather than milliseconds.  A value of 999 sends around 1000pps, and a value of 500 sends about 2000pps.  I get pretty much the same same results for -T gtod, nano, select.

**Comment by GrantEdwards** [comment]
To convert from ms to ns, you need to multiply by 1000000 instead of 1000:
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
diff --git a/src/tcpreplay_api.c b/src/tcpreplay_api.c
index fde27d6a..b6124358 100644
--- a/src/tcpreplay_api.c
+++ b/src/tcpreplay_api.c
@@ -200,7 +200,7 @@ tcpreplay_post_args(tcpreplay_t *ctx, int argc)
 
     if (HAVE_OPT(MAXSLEEP)) {
         options->maxsleep.tv_sec = OPT_VALUE_MAXSLEEP / 1000;
-        options->maxsleep.tv_nsec = (OPT_VALUE_MAXSLEEP % 1000) * 1000;
+        options->maxsleep.tv_nsec = (OPT_VALUE_MAXSLEEP % 1000) * 1000000;
     }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

**Comment by fklassen** [comment]
Thanks for the bug report and the patch. The patch fixes the issue.

**Comment by fklassen** [comment]
Fixed in Pr #460
